pkg/lib/config/saml.go

```diff
@@ -18,25 +18,27 @@ var _ = Schema.Add("SAMLServiceProviderConfig", `
 	"type": "object",
 	"additionalProperties": false,
 	"properties": {
-		"id": { "type": "string" }
+		"id": { "type": "string" },
+		"entity_id": { "type": "string" }
```

Comment:
Do we have an existing `format` that can be used here? If not, let's introduce a new one.

I think the format is

- If http: or https:, then it MUST BE an origin (that is, without userinfo, query, nor fragment)
- Otherwise it must be a urn:

Add tests for this new format.

Reply:
Oh, this was removed. Then the comment applies to the upcoming `audience`

Reply:
I don't think we need a strict check here.
1. It does not says it must be a origin. In fact, google does not expect a origin here, it must be `https://www.google.com/a/example.com/acs`.
2. The spec says it should be a uri, but does not says it must have scheme urn:.

I think we can simply keep this a string without any format check, because actually how to interpret `audience` is quite depends on implemntation of the SP, I prefer to keep the flexibility here.
And even if someone configured something here which is not an uri there is no harm to authgear.

Reply:
But the spec says it is a URI, so we at least must validate it is an absolute URI.

Reply:
I actually has a question on to what extend other SP implementations respect the spec. Just like what we see in auth0, non-uri values still passes validation.
But I think we can validate it is an uri first, and relax the validation if we see any problem.